Code/Editor/AzAssetBrowser/AzAssetBrowserRequestHandler.cpp

```diff
@@ -478,48 +593,87 @@ bool AzAssetBrowserRequestHandler::CanAcceptDragAndDropEvent(
         return false;
     }
 
-    bool canAcceptEvent = false;
+    return DecodeDragMimeData(event->mimeData());
+}
 
-    // Detects Source Asset Entries whose extensions are handled by a system
-    AzToolsFramework::AssetBrowser::AssetBrowserEntry::ForEachEntryInMimeData<SourceAssetBrowserEntry>(
-        event->mimeData(), [&](const AzToolsFramework::AssetBrowser::SourceAssetBrowserEntry* source) {
-            if (AssetBrowser::AssetBrowserSourceDropBus::HasHandlers(source->GetExtension()))
-            {
-                if (outSources.has_value())
-                {
-                    outSources.value()->push_back(source);
-                }
-                canAcceptEvent = true;
-            }
-        });
+bool AzAssetBrowserRequestHandler::DecodeDragMimeData(const QMimeData* mimeData,
+    AZStd::optional<AZStd::vector<const AzToolsFramework::AssetBrowser::ProductAssetBrowserEntry*>*> outProducts) const
```

Comment:
Why does this need to be an optional if it is a pointer?

It seems like this could just be `AZStd::vector<const AzToolsFramework::AssetBrowser::ProductAssetBrowserEntry*>*`

Reply:
Yeah, absolutely.  I was keeping it compatible with the prior API but I no longer see why it should be.  The main API doesn't have these optionals in it.  I'll just change it to a pointer, its so much simpler than optional, which seems to add unnecessary C++ Cruft when you know that its either going to be a specific type or nullptr